dali/pipeline/executor/executor.h

```diff
@@ -531,30 +537,32 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunGPU() {
   // iterations of a stage of the pipeline.
   CUDA_CALL(cudaEventSynchronize(gpu_stage_event_));
 
-  try {
     for (int i = 0; i < graph_->NumOp(OpType::GPU); ++i) {
       OpNode &op_node = graph_->Node(OpType::GPU, i);
-      typename WorkspacePolicy::template ws_t<OpType::GPU> ws =
-          WorkspacePolicy::template GetWorkspace<OpType::GPU>(gpu_idxs, *graph_, i);
-      auto parent_events = ws.ParentEvents();
+      try {
+        typename WorkspacePolicy::template ws_t<OpType::GPU> ws =
+            WorkspacePolicy::template GetWorkspace<OpType::GPU>(gpu_idxs, *graph_, i);
+        auto parent_events = ws.ParentEvents();
 
-      for (auto &event : parent_events) {
-        CUDA_CALL(cudaStreamWaitEvent(ws.stream(), event, 0));
-      }
+        for (auto &event : parent_events) {
+          CUDA_CALL(cudaStreamWaitEvent(ws.stream(), event, 0));
+        }
 
-      TimeRange tr("[Executor] Run GPU op " + op_node.instance_name,
-          TimeRange::knvGreen);
-      RunHelper(op_node, ws);
-      FillStats(gpu_memory_stats_, ws, "GPU_" + op_node.instance_name, gpu_memory_stats_mutex_);
-      if (ws.has_event()) {
-        CUDA_CALL(cudaEventRecord(ws.event(), ws.stream()));
+        TimeRange tr("[Executor] Run GPU op " + op_node.instance_name,
+            TimeRange::knvGreen);
+        RunHelper(op_node, ws);
+        FillStats(gpu_memory_stats_, ws, "GPU_" + op_node.instance_name, gpu_memory_stats_mutex_);
+        if (ws.has_event()) {
+          CUDA_CALL(cudaEventRecord(ws.event(), ws.stream()));
+        }
+      } catch (std::exception &e) {
+        HandleError(make_string("Error when executing GPU Operator ", op_node.op->name(),
+                                ", instance name: \"", op_node.instance_name, "\" encountered:\n",
+                                e.what()));
```

Comment:
I think this can be generic. As only CPU/Mixed/GPU changes...

Reply:
Done